component/header.go

```diff
@@ -32,28 +32,25 @@ func (*Header) Render() *vecty.HTML {
 // updatesHeader combines checkingForUpdates, noUpdates and updatesHeading
 // into one high level component.
 type updatesHeader struct {
-	RPs             []*model.RepoPresentation
-	CheckingUpdates bool
+	vecty.Core
+	RPs             []*model.RepoPresentation `vecty:"prop"`
+	CheckingUpdates bool                      `vecty:"prop"`
 }
 
-func (u updatesHeader) Render() []vecty.MarkupOrChild {
-	var ns []vecty.MarkupOrChild
-	// Show "Checking for updates..." while still checking.
-	if u.CheckingUpdates {
-		ns = append(ns, checkingForUpdates())
-	}
+func (u updatesHeader) Render() *vecty.HTML {
```

Comment:
I don't see a good reason not to just render this into its own element.

Reply:
There was a reason, although probably not a strong one. The original rendering code did not have a limitation of having each component render into a single HTML tag; it was able to render into multiple tags (i.e., into `[]*html.Node`).

The updates header happened to consist of multiple tags and didn't need a parent/container tag. I didn't want to add a tag for no reason just to be able to make this a true single vecty component. If performance is a reason, that's a different story, but, as I mentioned, I'll want to benchmark that.

Reply:
Did you get a chance to read the React reconciliation document?  I had hoped that would make it clear why this approach is a bad idea.  The change is not aimed a just making this a proper component, but more importantly to produce stable children in the parent.